pkg/kube/container/containers.go

```diff
@@ -152,3 +152,16 @@ func WithSecurityContext(context corev1.SecurityContext) Modification {
 		container.SecurityContext = &context
 	}
 }
+
+// WithAgentFlags takes a slice of envVars corresponding to
+// pairs of key-value agent startup flags and concatenates them
+// into a single string that is then passed as env variable AGENT_FLAGS
+func WithAgentFlags(envs ...corev1.EnvVar) Modification {
+	return func(container *corev1.Container) {
+		agentParams := ""
+		for _, variable := range envs {
+			agentParams += " -" + variable.Name + " " + variable.Value
+		}
+		container.Env = envvar.MergeWithOverride(container.Env, []corev1.EnvVar{{Name: "AGENT_FLAGS", Value: agentParams}})
+	}
+}
```

Comment:
nice implementation! I think however the containers package is not the correct place for this functionality. This specific function is now only exists for use with the agent (especially with the hard coded `AGENT_FLAGS` env var name)

To me this is something that we could leverage both in the community and enterprise operator, but I think it belongs in an `agents` or `agent` package, rather than `container`. 

To me it makes more sense to have the following structure

 `[]customType` -> `[]corev1.EnvVar` -> `container.WithEnvs(agentEnvs)`

rather than including this logic in this package.

What do you think?